apollo-router/src/configuration/mod.rs

```diff
@@ -177,6 +177,46 @@ pub struct Configuration {
     dev: Option<bool>,
 }
 
+impl<'de> serde::Deserialize<'de> for Configuration {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        // This intermediate structure will allow us to deserialize a Configuration
+        // yet still exercise the Configuration validation function
+        #[derive(Deserialize, Default)]
+        struct AdHocConfiguration {
```

Comment:
I'm not sure to understand why we have this similar struct ? Why not deserializer in `Configuration` and call `conf.validate()` inside this `deserialize` method ?

Reply:
I'm writing a manual implementation of Deserialize for Configuration to force validation, I don't think I can call deserialize (that would be me recursing on myself 🤔)

Reply:
Oh yes indeed. hmmm...

Reply:
if you're going that way, it might be simpler to have a `UnverifiedConfiguration` implementing deserialize with attributes, then a `Configuration` that try_froms the first one and cannot be constructed in any other way

Reply:
This would unfortunately change the public api :(

Reply:
Not necessarily, it could be Configuration publicly, and FinalConfiguration2ForRealThisTime for the internal one 😄